### PR TITLE
NC-558 Vue 1.0

### PR DESCRIPTION
**Pull request by Joeper214**
http://vuejs.org/2015/10/26/1.0.0-release/

Upgrade to Vue 1.0.0

NOTES:
-------------------------------------------
To upgrade/fix your implementations to Vue 1.0.0.
Please branch out to this branch only (not in develop).

And please try to create your own branch before doing the refactor.

After that, please merge/ create PR of your work into this branch.

Goal
-----

- [x] Serverspec generator
- [x] users_admin
- [x] ELB
- [x] S3
- [x] server state button
- [x] RDS
- [x] editor monitoring
- [x] vue helper

Update process
----------

```sh
cd YOUR_SKYHOPPER_DIRECTORY
# bundle install
# bower install
# cd frontend/
# npm i
# gulp tsd
# gulp ts
cd ..
```

Thank you!.

@pocke 
@root-beer

### CODE REVIEW

File: app/assets/javascripts/clients.js [hunk 1]
```diff
@@ -12,39 +12,41 @@
...
     template: '#grid-template',
     replace: true,
-    props: ['data', 'columns', 'filter-key'],
+    props: {
+      data: Array,
+      columns: Array,
+      filterKey: String
+    },
     data: function () {
+      var sortOrders = {}
```
Comment (skyarch-networks) [inline app/assets/javascripts/clients.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/clients.js [hunk 2]
```diff
@@ -12,39 +12,41 @@
...
+      this.columns.forEach(function (key) {
+        sortOrders[key] = 1
```
Comment (skyarch-networks) [inline app/assets/javascripts/clients.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/clients.js [hunk 3]
```diff
@@ -12,39 +12,41 @@
...
+      })
```
Comment (skyarch-networks) [inline app/assets/javascripts/clients.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/clients.js [hunk 4]
```diff
@@ -101,7 +102,12 @@
...
-    clientIndex();
-  });
+  var demo = new Vue({
+    el: '#indexElement',
+    data: {
+      searchQuery: '',
+      gridColumns: ['code','name', 'id'],
+      gridData: []
+    }
+  })
```
Comment (skyarch-networks) [inline app/assets/javascripts/clients.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/infrastructures.js [hunk 5]
```diff
@@ -2005,14 +2005,19 @@
...
     template: '#grid-template',
     replace: true,
-    props: ['data', 'columns', 'filter-key'],
+    props: {
+      data: Array,
+      columns: Array,
+      filterKey: String
+    },
     data: function () {
+      var sortOrders = {}
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/infrastructures.js [hunk 6]
```diff
@@ -2005,14 +2005,19 @@
...
+      this.columns.forEach(function (key) {
+        sortOrders[key] = 1
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/infrastructures.js [hunk 7]
```diff
@@ -2005,14 +2005,19 @@
...
+      })
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/infrastructures.js [hunk 8]
```diff
@@ -2271,8 +2254,25 @@
...
+      gridData: []
+    },
+    created: function(){
+        if (queryString.project_id >3)
+          this.gridColumns = ['stack_name','region', 'keypairname', 'created_at', 'status', 'id'];
+        else
+          this.gridColumns = ['stack_name','region', 'keypairname', 'id'];
+    },
+
+  })
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/projects.js [hunk 9]
```diff
@@ -12,39 +12,35 @@
...
     template: '#grid-template',
     replace: true,
-    props: ['data', 'columns', 'filter-key'],
+    props: {
+      data: Array,
+      columns: Array,
+      filterKey: String
+    },
     data: function () {
+      var sortOrders = {}
```
Comment (skyarch-networks) [inline app/assets/javascripts/projects.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/projects.js [hunk 10]
```diff
@@ -12,39 +12,35 @@
...
+      this.columns.forEach(function (key) {
+        sortOrders[key] = 1
```
Comment (skyarch-networks) [inline app/assets/javascripts/projects.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/projects.js [hunk 11]
```diff
@@ -12,39 +12,35 @@
...
+      })
```
Comment (skyarch-networks) [inline app/assets/javascripts/projects.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/user_index.js [hunk 12]
```diff
@@ -20,32 +19,31 @@
...
     template: '#grid-template',
     replace: true,
-    props: ['data', 'columns', 'filter-key'],
+    props: {
+      data: Array,
+      columns: Array,
+      filterKey: String
+    },
     data: function () {
+      var sortOrders = {}
```
Comment (skyarch-networks) [inline app/assets/javascripts/user_index.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/user_index.js [hunk 13]
```diff
@@ -20,32 +19,31 @@
...
+      this.columns.forEach(function (key) {
+        sortOrders[key] = 1
```
Comment (skyarch-networks) [inline app/assets/javascripts/user_index.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/user_index.js [hunk 14]
```diff
@@ -20,32 +19,31 @@
...
+      })
```
Comment (skyarch-networks) [inline app/assets/javascripts/user_index.js]: __[jshint]__
*  Missing semicolon.

File: app/assets/javascripts/infrastructures.js [hunk 15]
```diff
@@ -2005,33 +2005,31 @@
...
+    props: {
+      data: Array,
+      columns: Array,
+      filterKey: String
+    },
     data: function () {
+      var sortOrders = {};
+      this.columns.forEach(function (key) {
+        sortOrders[key] = 1;
+      })
```
Comment (skyarch-networks) [inline app/assets/javascripts/infrastructures.js]: __[jshint]__
*  Missing semicolon.

File: frontend/src/helper.ts [hunk 16]
```diff
@@ -26,7 +26,7 @@ Vue.component('bootstrap-tooltip', {
       required: true,
     },
   },
-  template: '<span data-toggle="tooltip" data-original-title="{{title}}"><content></content></span>',
+  template: '<span data-toggle="tooltip" data-original-title="{{title}}"><content class="content"><slot name="content"></slot></content></span>',
```
Comment (skyarch-networks) [inline frontend/src/helper.ts]: __[tslint]__
* exceeds maximum line length of 140

**Comment by Joeper214** [comment]
I think I'm done refactoring the codes. and there are still errors remaining as mentioned in slack. I will assign it to @root-beer . Thanks!

**Comment by pocke** [comment]
I'll try this branch

**Comment by Joeper214** [comment]
Thanks. :bow:

**Comment by Joeper214** [comment]
Please checkout Serverspec Generator page. I think it's not yet refactored to Vue 1.0

**Comment by pocke** [comment]
OK, I'll try this after #104 is merged

**Comment by pocke** [comment]
Migration tips
========



v-on
-----

in Vim

`%s/\v\Vv-on="\v(\w+): ([^"]+)"/@\1="\2"/gc`


v-attr
-----

in Vim

`%s/\vv-attr\="(\w+): ([^"]+)"/:\1="\2"/gc`


v-repeat
-----

in Vim

`%s/\vv-repeat\="(\w+):\s+((\w|.)+)"/v-for="\1 in \2"/gc`

**Comment by Joeper214** [comment]
That's Great! Only if I were using vim.  :(

**Comment by pocke** [comment]
I think this task was completed.

Please review :bow: @Joeper214 @root-beer

**Comment by Joeper214** [comment]
LGTM!

**Comment by pocke** [comment]
Thanks!

**Pull request merged**